bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/filters/BigtableWhileMatchResultScannerAdapter.java

```diff
@@ -53,32 +59,44 @@ public BigtableWhileMatchResultScannerAdapter(ResponseAdapter<FlatRow, Result> r
    * <p>adapt.</p>
    *
    * @param bigtableResultScanner a {@link com.google.cloud.bigtable.grpc.scanner.ResultScanner} object.
+   * @param span A parent span for the scan that needs to be closed when the 
    * @return a {@link org.apache.hadoop.hbase.client.ResultScanner} object.
    */
   public ResultScanner adapt(
-      final com.google.cloud.bigtable.grpc.scanner.ResultScanner<FlatRow> bigtableResultScanner) {
+      final com.google.cloud.bigtable.grpc.scanner.ResultScanner<FlatRow> bigtableResultScanner,
+      final Span span) {
     return new AbstractClientScanner() {
       @Override
       public Result next() throws IOException {
         FlatRow row = bigtableResultScanner.next();
         if (row == null) {
           // Null signals EOF.
+          closeSpan();
           return null;
         }
 
         if (!hasMatchingLabels(row)) {
+          close();
           return null;
         }
 
         return rowAdapter.adaptResponse(row);
       }
 
+      protected void closeSpan() {
+        try (NonThrowingCloseable s = Tracing.getTracer().withSpan(span)) {
```

Comment:
s/NonThrowingCloseable/Closeable

Reply:
NonThrowingCloseable does not throw exceptions, Closable does.  I can do a try/catch, which does nothing...